enderio-base/src/main/java/com/enderio/base/common/item/darksteel/DarkSteelAxeItem.java

```diff
@@ -71,6 +112,48 @@ private boolean hasFork(ItemStack stack) {
         return DarkSteelUpgradeable.hasUpgrade(stack, ForkUpgrade.NAME);
     }
 
+    private void fellTree(Level level, BlockPos pos, Set<BlockPos> checkedPos, Set<BlockPos> toChop, int maxBlocks, Block targetBock) {
+        if(toChop.size() >= maxBlocks || checkedPos.contains(pos)) {
+            return;
+        }
+        checkedPos.add(pos);
+        BlockState checkState = level.getBlockState(pos);
+        if (checkState.is(targetBock)) {
+            toChop.add(pos);
+
+            Set<BlockPos> toCheck = new HashSet<>();
+            surrounding(toCheck, pos);
+            surrounding(toCheck, pos.above());
+            toCheck.add(pos.above());
+            for(BlockPos newPos : toCheck) {
+                fellTree(level, newPos, checkedPos, toChop, maxBlocks, targetBock);
+            }
+        }
+    }
+
+    private void surrounding(Set<BlockPos> res, BlockPos pos) {
+        for(Direction dir : Direction.values()) {
+            if(dir != DOWN && dir != UP) {
+                res.add(pos.relative(dir));
+            }
+        }
+        res.add(pos.north().east());
+        res.add(pos.north().west());
+        res.add(pos.south().east());
+        res.add(pos.south().west());
```

Comment:
suggestion:
I think this is more readable
```suggestion
        for (int x = -1; x <= 1; x++) {
            for (int z = -1; z <= 1; z++) {
                if (x != 0 || z != 0) {
                  res.add(pos.offset(x, 0, z));
                }
            }
        }
```

Reply:
or:
```suggestion
        for (int x = -1; x <= 1; x++) {
            for (int z = -1; z <= 1; z++) {
                res.add(pos.offset(x, 0, z));
            }
        }
        res.remove(pos);
```